### PR TITLE
Refresh site name textview on resume Fixes #3607

### LINKED ISSUE #3607: Site Settings: Updating site title isn't reflected in the rest of the app without reselecting the site
If you update your site's title and then return to the main menu, you'll still see the old title at the top of the page until you either tap Switch Site and re-select it, or close and re-launch the app.

Screencast: https://cloudup.com/c-6J8y6dzmM

### PR DESCRIPTION
**Pull request by hypest**
The site name may have been changed via the Settings so, need to update the relevant UI upon returning.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java [hunk 1]
```diff
@@ -86,6 +86,12 @@ public void onPause() {
     @Override
     public void onResume() {
         super.onResume();
+
+        // Site name may have changed (e.g. via Settings and returning to this Fragment) so update the UI
+        if (mBlog != null && !mBlogTitleTextView.getText().equals(mBlog.getBlogName())) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java]: This is quite different from how the blog title is set in [refreshBlogDetails](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/3607-site-name-refresh/WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java#L324-L334).

Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java]: Hmm, indeed. I used this method: https://github.com/wordpress-mobile/WordPress-Android/blob/issue/3607-site-name-refresh/WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java#L370-L372

Happy to change it to the one you've highlight though if that's more robust! I'm sure i'm missing some background on this one.

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java]: > Hmm, indeed. I used this method:

Yuck. Seems that we should have a common routine which displays the blog name & url using the same technique as `refreshBlogDetails` (which appears to deal with the rare situation that the blog name is empty).

Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java]: I can make that happen, cool :)

**Comment by nbradbury** [comment]
I just tested this, and the blog name change isn't being shown in My Site. The problem is that `updateBlogNameView()` relies on the existing instance of `mBlog`, which isn't updated after the user returns from settings.

**Comment by nbradbury** [comment]
This doesn't need to be changed unless you feel like it, just wanted to mention it: to me it would make sense if `refreshBlogDetails()` accepted no parameters and instead loaded the blog object from the db based on `mBlogLocalId` - ie:
```
private void refreshBlogDetails() {
    if (!isAdded()) {
        return;
    }

    final Blog blog = WordPress.getBlog(mBlogLocalId);
    if (blog == null) {
    . . .
```
Doing that would require changing [this line](https://github.com/wordpress-mobile/WordPress-Android/commit/a12eb1ef442eaaea0033969d7adfcb9a8c1ad6e7#diff-57a50fd5aaef95ac8ac5d2ebb2eb559eR108) as well. I think `WordPress.wpDB.isLocalBlogIdInDatabase(mBlogLocalId)` would suffice.

**Comment by hypest** [comment]
Re: the parameter in `refreshBlogDetails()`, that method is used both in places where the blog is already available (see https://github.com/wordpress-mobile/WordPress-Android/commit/a12eb1ef442eaaea0033969d7adfcb9a8c1ad6e7#diff-57a50fd5aaef95ac8ac5d2ebb2eb559eR69) and in other where the blog is loaded just before the `refreshBlogDetails()` call. So, it made more sense to avoid a reloading the data and instead just provide it through the parameter.

This also slightly bauds better with the attempt to reduce the dependence on Fragment state variables.

**Comment by nbradbury** [comment]
@hypest This looks good! :shipit:

**Pull request merged**